diff -Naur src/mmpi_asynchronous.h ../../mikempi-1.0.102/src/mmpi_asynchronous.h
--- src/mmpi_asynchronous.h	2006-09-13 10:52:42.000000000 +0200
+++ ../../mikempi-1.0.102/src/mmpi_asynchronous.h	2006-09-06 16:25:06.000000000 +0200
@@ -27,7 +27,7 @@
 /* Standard Header files */
 #include <mpi.h>
 #include <Python.h>
-#include <numpy/arrayobject.h>
+#include <numpy/oldnumeric.h>
 #include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -Naur src/mmpi_cartesian.h ../../mikempi-1.0.102/src/mmpi_cartesian.h
--- src/mmpi_cartesian.h	2006-09-13 10:52:42.000000000 +0200
+++ ../../mikempi-1.0.102/src/mmpi_cartesian.h	2006-09-06 16:25:58.000000000 +0200
@@ -28,7 +28,7 @@
 /* Standard Header files */
 #include <mpi.h>
 #include <Python.h>
-#include <numpy/arrayobject.h>
+#include <numpy/oldnumeric.h>
 #include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -Naur src/mmpi_collective.h ../../mikempi-1.0.102/src/mmpi_collective.h
--- src/mmpi_collective.h	2006-09-13 10:52:42.000000000 +0200
+++ ../../mikempi-1.0.102/src/mmpi_collective.h	2006-09-06 16:26:04.000000000 +0200
@@ -20,7 +20,7 @@
 
 /* Standard Header files */
 #include <Python.h>
-#include <numpy/arrayobject.h>
+#include <numpy/oldnumeric.h>
 #include <mpi.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -Naur src/mmpi_comm.h ../../mikempi-1.0.102/src/mmpi_comm.h
--- src/mmpi_comm.h	2006-09-13 10:52:42.000000000 +0200
+++ ../../mikempi-1.0.102/src/mmpi_comm.h	2006-09-06 16:26:11.000000000 +0200
@@ -22,7 +22,7 @@
 /* Standard Header files */
 #include <mpi.h>
 #include <Python.h>
-#include <numpy/arrayobject.h>
+#include <numpy/oldnumeric.h>
 
 #include <sys/types.h>
 #include <sys/stat.h>
diff -Naur src/mmpi_error.h ../../mikempi-1.0.102/src/mmpi_error.h
--- src/mmpi_error.h	2006-09-13 10:52:42.000000000 +0200
+++ ../../mikempi-1.0.102/src/mmpi_error.h	2006-09-06 16:26:25.000000000 +0200
@@ -21,7 +21,7 @@
 
 /* Standard Header files */
 #include <Python.h>
-#include <numpy/arrayobject.h>
+#include <numpy/oldnumeric.h>
 #include <mpi.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -Naur src/mmpi_graph.h ../../mikempi-1.0.102/src/mmpi_graph.h
--- src/mmpi_graph.h	2006-09-13 10:52:42.000000000 +0200
+++ ../../mikempi-1.0.102/src/mmpi_graph.h	2006-09-06 16:26:37.000000000 +0200
@@ -27,7 +27,7 @@
 /* Standard Header files */
 #include <mpi.h>
 #include <Python.h>
-#include <numpy/arrayobject.h>
+#include <numpy/oldnumeric.h>
 #include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -Naur src/mmpi_synchronous.h ../../mikempi-1.0.102/src/mmpi_synchronous.h
--- src/mmpi_synchronous.h	2006-09-13 10:52:42.000000000 +0200
+++ ../../mikempi-1.0.102/src/mmpi_synchronous.h	2006-09-06 16:26:42.000000000 +0200
@@ -20,7 +20,7 @@
 
 /* Standard Header files */
 #include <Python.h>
-#include <numpy/arrayobject.h>
+#include <numpy/oldnumeric.h>
 #include <mpi.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -Naur src/mmpi_timing.h ../../mikempi-1.0.102/src/mmpi_timing.h
--- src/mmpi_timing.h	2006-09-13 10:52:42.000000000 +0200
+++ ../../mikempi-1.0.102/src/mmpi_timing.h	2006-09-06 16:26:49.000000000 +0200
@@ -22,7 +22,7 @@
 /* Standard Header files */
 #include <mpi.h>
 #include <Python.h>
-#include <numpy/arrayobject.h>
+#include <numpy/oldnumeric.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -Naur src/mmpi_utilities.h ../../mikempi-1.0.102/src/mmpi_utilities.h
--- src/mmpi_utilities.h	2006-09-13 10:52:42.000000000 +0200
+++ ../../mikempi-1.0.102/src/mmpi_utilities.h	2006-09-06 16:27:01.000000000 +0200
@@ -25,7 +25,7 @@
 #define MPIMODULE_UTILITIES_H
 
 #include <Python.h>
-#include <numpy/arrayobject.h>
+#include <numpy/oldnumeric.h>
 
 #include "mmpi_globals.h"
 
