diff -ur ./setup.py ../../mikempi-1.1.134/setup.py
--- ./setup.py	2007-04-27 11:00:47.000000000 +0200
+++ ../../mikempi-1.1.134/setup.py	2007-04-27 10:52:41.000000000 +0200
@@ -15,17 +15,57 @@
 # are possible values for "array_prefix"
 array_prefix = ''
 mpicc_bin = 'mpicc'
+use_mpe = True
 
 argv_replace = []
 
+
 for arg in sys.argv:
     if arg.startswith('--with-array='):
         array_prefix = arg.split('=', 1)[1]
     elif arg.startswith('--with-mpicc='):
         mpicc_bin = arg.split('=', 1)[1]
+    elif arg.startswith('--without-mpe'):
+        use_mpe = False
     else:
         argv_replace.append(arg)
 
+#try to autodetect array library if it is not specified on the cmd line
+if array_prefix == "":
+	numpy_available = False
+	numeric_available = False
+	numarray_available = False
+	
+	#Check if numpy exists
+	try:
+		import numpy
+		numpy_available = True
+	except: pass
+
+	#Check if Numeric exists
+	try:
+		import Numeric
+		numeric_available = True
+	except: pass
+
+	#Check if Numarray is available
+	try:
+		import Numarray
+		numarray_available = True
+	except: pass
+
+	#Numpy is the preferred array library.
+	if numpy_available:
+		array_prefix = "numpy"
+	elif numeric_available:
+		array_prefix = "Numeric"
+	elif numarray_available:
+		array_prefix = "Numarray"
+	else:
+		raise "Unable to find any supported array package. Please install numpy, Numeric or Numarray"
+		
+	
+
 sys.argv = argv_replace
 
 # this will handle most of the compiler calls, using mpicc_bin
@@ -48,14 +88,25 @@
 # now go on and do the actual setup
 
 SOURCE = ["src/mmpi_module.c"]
-INCLUDES = [os.path.join( get_python_inc(), array_prefix )
-            ]
+INCLUDES = []
+DEFINES = []
+
+if array_prefix.lower() == "numpy":
+    import numpy
+    INCLUDES += [ os.path.join( numpy.__path__[0], "core", "include", "numpy") ]
+    DEFINES += [("NUMPY", None)]
+else:
+    INCLUDES += [os.path.join( get_python_inc(), array_prefix ) ]
 
 MPESOURCE = ["src/mpe_module.c"]
 LIBRARIES = ["mpe"]
 
 print 'Setting MMPI setup with Python ',sys.executable,', version ',sys.version
 
+extensions = [Extension('_mpi', SOURCE, include_dirs=INCLUDES, define_macros=DEFINES)]
+if use_mpe:
+    extensions += [Extension('_mpe', MPESOURCE, include_dirs=INCLUDES, libraries=LIBRARIES, define_macros=DEFINES)] 
+
 setup(name='MMPI',
       version='1.0',
       description='Lightweight MPI module for Python 2.4+',
@@ -64,7 +115,6 @@
       author_email='steder@gmail.com',
       packages=['mpi'],
       ext_package='mpi',
-      ext_modules=[Extension('_mpi', SOURCE, include_dirs=INCLUDES),
-                   Extension('_mpe', MPESOURCE, include_dirs=INCLUDES, libraries=LIBRARIES)],
+      ext_modules=extensions,
       )
 
Only in .: setup.py.orig
Only in .: setup.py.rej
Only in ../../mikempi-1.1.134/: .setup.py.swp
diff -ur ./src/mmpi_asynchronous.h ../../mikempi-1.1.134/src/mmpi_asynchronous.h
--- ./src/mmpi_asynchronous.h	2006-07-12 03:10:24.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_asynchronous.h	2007-04-27 10:29:29.000000000 +0200
@@ -27,7 +27,13 @@
 /* Standard Header files */
 #include "mpi.h"
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 #include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -ur ./src/mmpi_cartesian.h ../../mikempi-1.1.134/src/mmpi_cartesian.h
--- ./src/mmpi_cartesian.h	2006-07-12 03:10:24.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_cartesian.h	2007-04-27 10:29:55.000000000 +0200
@@ -28,7 +28,13 @@
 /* Standard Header files */
 #include "mpi.h"
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 #include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -ur ./src/mmpi_collective.h ../../mikempi-1.1.134/src/mmpi_collective.h
--- ./src/mmpi_collective.h	2006-07-12 03:22:11.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_collective.h	2007-04-27 10:30:04.000000000 +0200
@@ -20,7 +20,13 @@
 
 /* Standard Header files */
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 #include "mpi.h"
 #include <stdio.h>
 #include <stdlib.h>
diff -ur ./src/mmpi_comm.h ../../mikempi-1.1.134/src/mmpi_comm.h
--- ./src/mmpi_comm.h	2006-07-12 03:10:24.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_comm.h	2007-04-27 10:30:18.000000000 +0200
@@ -22,7 +22,13 @@
 /* Standard Header files */
 #include "mpi.h"
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ur ./src/mmpi_error.h ../../mikempi-1.1.134/src/mmpi_error.h
--- ./src/mmpi_error.h	2006-07-12 03:10:24.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_error.h	2007-04-27 10:30:30.000000000 +0200
@@ -21,7 +21,13 @@
 
 /* Standard Header files */
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 #include "mpi.h"
 #include <stdio.h>
 #include <stdlib.h>
diff -ur ./src/mmpi_graph.h ../../mikempi-1.1.134/src/mmpi_graph.h
--- ./src/mmpi_graph.h	2006-07-12 03:10:24.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_graph.h	2007-04-27 10:30:42.000000000 +0200
@@ -27,7 +27,13 @@
 /* Standard Header files */
 #include "mpi.h"
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 #include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -ur ./src/mmpi_intercomm.h ../../mikempi-1.1.134/src/mmpi_intercomm.h
--- ./src/mmpi_intercomm.h	2006-07-12 03:10:24.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_intercomm.h	2007-04-27 10:30:48.000000000 +0200
@@ -22,7 +22,13 @@
 /* Standard Header files */
 #include "mpi.h"
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ur ./src/mmpi_io.c ../../mikempi-1.1.134/src/mmpi_io.c
--- ./src/mmpi_io.c	2006-07-16 08:04:04.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_io.c	2007-04-27 10:30:57.000000000 +0200
@@ -26,7 +26,13 @@
 */
 /* Standard Header files */
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 #include "mpi.h"
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ur ./src/mmpi_module.c ../../mikempi-1.1.134/src/mmpi_module.c
--- ./src/mmpi_module.c	2006-07-12 03:10:24.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_module.c	2007-04-27 10:31:04.000000000 +0200
@@ -32,7 +32,13 @@
 */
 /* Standard Header files */
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 #include "mpi.h"
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ur ./src/mmpi_timing.h ../../mikempi-1.1.134/src/mmpi_timing.h
--- ./src/mmpi_timing.h	2006-07-12 03:10:24.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_timing.h	2007-04-27 10:31:10.000000000 +0200
@@ -22,7 +22,13 @@
 /* Standard Header files */
 #include "mpi.h"
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -ur ./src/mmpi_types.h ../../mikempi-1.1.134/src/mmpi_types.h
--- ./src/mmpi_types.h	2006-07-12 03:10:24.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_types.h	2007-04-27 10:31:23.000000000 +0200
@@ -22,7 +22,13 @@
 /* Standard Header files */
 #include "mpi.h"
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ur ./src/mmpi_utilities.h ../../mikempi-1.1.134/src/mmpi_utilities.h
--- ./src/mmpi_utilities.h	2006-07-12 03:22:11.000000000 +0200
+++ ../../mikempi-1.1.134/src/mmpi_utilities.h	2007-04-27 10:26:01.000000000 +0200
@@ -25,7 +25,12 @@
 #define MPIMODULE_UTILITIES_H
 
 #include "Python.h"
+
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
 
 #include "mmpi_globals.h"
 
Only in ../../mikempi-1.1.134/src: .mmpi_utilities.h.swo
Only in ../../mikempi-1.1.134/src: .mmpi_utilities.h.swp
diff -ur ./src/mpe_module.c ../../mikempi-1.1.134/src/mpe_module.c
--- ./src/mpe_module.c	2006-07-12 03:10:24.000000000 +0200
+++ ../../mikempi-1.1.134/src/mpe_module.c	2007-04-27 10:31:40.000000000 +0200
@@ -28,7 +28,13 @@
 #include "mpi.h"
 #include "mpe.h"
 #include "Python.h"
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
 #include "arrayobject.h"
+#endif
+
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -ur ./src/template.c.ignore ../../mikempi-1.1.134/src/template.c.ignore
--- ./src/template.c.ignore	2006-01-31 21:53:58.000000000 +0100
+++ ../../mikempi-1.1.134/src/template.c.ignore	2007-04-27 10:34:09.000000000 +0200
@@ -23,7 +23,13 @@
 
 /* Standard Header files */
 #include <Python.h>
-#include <Numeric/arrayobject.h>
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
+#include "arrayobject.h"
+#endif
+
+
 #include <mpi.h>
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ur ./src/template.h.ignore ../../mikempi-1.1.134/src/template.h.ignore
--- ./src/template.h.ignore	2006-01-31 21:53:58.000000000 +0100
+++ ../../mikempi-1.1.134/src/template.h.ignore	2007-04-27 10:34:14.000000000 +0200
@@ -20,7 +20,13 @@
 
 /* Standard Header files */
 #include <Python.h>
-#include <Numeric/arrayobject.h>
+#ifdef NUMPY
+#include "oldnumeric.h"
+#else
+#include "arrayobject.h"
+#endif
+
+
 #include <mpi.h>
 #include <stdio.h>
 #include <stdlib.h>
